components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx

```diff
@@ -188,62 +193,121 @@ export const Nfts = (props: Props) => {
+    // add NFTs user has marked as NFT if they are not in the list
+    // to avoid duplicates
+    const fullSpamList = [
+      ...simpleHashList,
+      ...userMarkedSpamNfts.filter(
+        (nft) => !simpleHashListIds.includes(getAssetIdKey(nft))
+      )
+    ]
 
+    return [
+      fullSpamList,
+      fullSpamList.map((nft) => getAssetIdKey(nft))
+    ]
+  }, [userMarkedSpamNfts, simpleHashSpamNfts, hiddenNftsIds, userNonSpamNftIds])
+
+  const [sortedNfts, sortedHiddenNfts, sortedSpamNfts] = React.useMemo(() => {
     if (searchValue === '') {
       return [
-        nftList.slice().sort(compareFn),
-        hiddenNfts.slice().sort(compareFn)
+        userNonSpamNfts.slice().sort(compareFn),
+        hiddenNfts.slice().sort(compareFn),
+        allSpamNfts.slice().sort(compareFn)
       ]
     }
 
     return [
-      nftList.filter(searchNfts).sort(compareFn),
-      hiddenNfts.filter(searchNfts).sort(compareFn)
+      userNonSpamNfts.filter(searchNfts).sort(compareFn),
+      hiddenNfts.filter(searchNfts).sort(compareFn),
+      allSpamNfts.filter(searchNfts).sort(compareFn)
     ]
-  }, [searchValue, nftList, hiddenNfts, searchNfts])
+  }, [searchValue, userNonSpamNfts, hiddenNfts, allSpamNfts, searchNfts])
 
   const tabOptions = React.useMemo(() => {
-    const tabOptions: TabOption[] = [
+    const tabOptions: NftTabOption[] = [
       {
         id: 'nfts',
         label: getLocale('braveNftsTab'),
-        labelSummary: nftList.length || undefined
+        labelSummary: userNonSpamNfts.length || undefined
       },
       {
         id: 'hidden',
         label: getLocale('braveNftsTabHidden'),
         labelSummary: hiddenNfts.length || undefined
+      },
+      {
+        id: 'spam',
+        label: getLocale('braveNftsTabSpam'),
+        labelSummary: !isLoadingSimpleHashList ? allSpamNfts.length : ''
       }
     ]
 
     return tabOptions
-  }, [nftList, hiddenNfts])
-
-
-  const renderedList = React.useMemo(() => {
-    return selectedTab === 'nfts' ? sortedNfts : sortedHiddenNfts
-  }, [selectedTab, sortedNfts, sortedHiddenNfts])
+  }, [nftList, hiddenNfts, allSpamNfts, isLoadingSimpleHashList])
+
+
+  const getRenderedList = () => {
+    switch (selectedTab) {
+      case 'nfts':
+        return sortedNfts;
+      case 'hidden':
+        return sortedHiddenNfts;
+      case 'spam':
+        return sortedSpamNfts;
+      default:
+        return sortedNfts;
+    }
+  }
 
   // Returns a list of assets based on provided account
   const getFilteredNftsByAccount = React.useCallback(
     (account: BraveWallet.AccountInfo) => {
-      return renderedList.filter(
+      return getRenderedList().filter(
         (nft) =>
           nft.coin === account.accountId.coin &&
           new Amount(
             getBalance(account.accountId, nft, tokenBalancesRegistry)
           ).gte('1')
       )
     },
-    [renderedList, tokenBalancesRegistry]
+    [getRenderedList, tokenBalancesRegistry]
```

Comment:
getRenderedList will cause this callback to always be recomputed. Either unwrap this function too, or wrap getRenderedList again

Reply:
I have wrapped getRenderedList using useMemo and renamed it to `renderedList`